src/rpc/RPCHelpers.cpp

```diff
@@ -1478,4 +1478,210 @@ specifiesCurrentOrClosedLedger(boost::json::object const& request)
     return false;
 }
 
+std::variant<ripple::uint256, Status>
+getNFTID(boost::json::object const& request)
+{
+    if (!request.contains(JS(nft_id)))
+        return Status{Error::rpcINVALID_PARAMS, "missingTokenid"};
+
+    if (!request.at(JS(nft_id)).is_string())
+        return Status{Error::rpcINVALID_PARAMS, "tokenidNotString"};
+
+    ripple::uint256 tokenid;
+    if (!tokenid.parseHex(request.at(JS(nft_id)).as_string().c_str()))
+        return Status{Error::rpcINVALID_PARAMS, "malformedCursor"};
+
+    return tokenid;
+}
+
+// TODO - this function is long and shouldn't be responsible for as much as it
+// is. Split it out into some helper functions.
```

Comment:
You could split this into pre-processing and post-processing

Reply:
totally - there's a lot of room to improve this. this is mostly just the entirety of the old account_tx driver. what do you think about leaving such a refactor for later? I want to keep the footprint of this PR small :)

Reply:
It's up to you whether you want to refactor now or later. This function is a bit hairy and the hairiness often leads to bugs, especially since we don't have unit tests for the handlers. If we refactor into helpers, we could unit test the helpers. But up to you, will let you decide.